pkg/controllers/provisioningv2/rke2/machineprovision/args.go

```diff
@@ -115,30 +121,37 @@ func (h *handler) getArgsEnvAndStatus(meta metav1.Object, data data.Object, args
 			fmt.Sprintf("--driver=%s", driver),
 			fmt.Sprintf("--custom-install-script=/run/secrets/machine/value"))
 
-		rancherCluster, err := h.rancherClusterCache.Get(meta.GetNamespace(), meta.GetLabels()[capi.ClusterLabelName])
+		rancherCluster, err := h.rancherClusterCache.Get(infraObj.meta.GetNamespace(), infraObj.meta.GetLabels()[capi.ClusterLabelName])
 		if err != nil {
 			return driverArgs{}, err
 		}
 		cmd = append(cmd, toArgs(driver, args, rancherCluster.Status.ClusterName)...)
+		jobBackoffLimit = 1
```

Comment:
From what I see we've used backoff limit of 0 in create situation and now going with 1. Is this intended?

Reply:
No, this is a mistake. Will fix.

Reply:
Fixed.